elephant/spike_train_surrogates.py

```diff
@@ -111,23 +111,23 @@ def dither_spikes(spiketrain, dither, n=1, decimals=None, edges=True):
     # Main: generate the surrogates
     surr = data.reshape((1, len(data))) + 2 * dither * np.random.random_sample(
         (n, len(data))) - dither
-
+    print(surr)
     # Round the surrogate data to decimal position, if requested
     if decimals is not None:
         surr = surr.round(decimals)
 
     if edges is False:
         # Move all spikes outside [spiketrain.t_start, spiketrain.t_stop] to
         # the range's ends
-        surr = np.minimum(np.maximum(surr.base,
-            (spiketrain.t_start / spiketrain.units).base),
-            (spiketrain.t_stop / spiketrain.units).base) * spiketrain.units
+        surr = np.minimum(np.maximum(surr.simplified.magnitude,
+            spiketrain.t_start.simplified.magnitude),
+            spiketrain.t_stop.simplified.magnitude) * pq.s
```

Comment:
This solution looks clean and relatively compact, but it forces all numerical operations to be held in seconds. What about applying the unit of original spike train from the beginning: 
```
surr = np.minimum(np.maximum(surr.magnitude,
    (spiketrain.t_start.rescale(spiketrain.units).magnitude)),
    (spiketrain.t_stop.rescale(spiketrain.units).magnitude)) * spiketrain.units
```
Something has to be rescaled, if spiketrain and its t_start and t_stop have different units, but hard-coding any arbitrary units always seems a bit worrisome.

Reply:
The output spiketrains are in the end all re-scaled to the input for the units, honestly for the computations I prefer to have explicitely fixed units everywhere, which are not arbitrary but they match the simplified function